examples/nlp/lasertagger/README.md

```diff
@@ -0,0 +1,130 @@
+running example.
+
+You can run all of the steps with
+
+```
+sh run_summarization.sh
+```
+
+after setting the paths in the beginning of the script.
+
+**Note:** Text should be tokenized with spaces separating the tokens before applying LaserTagger.
+
+### 1. Phrase Vocabulary Optimization
+
+The MSR Abstractive Summarization dataset is in the `data` folder. This has been pretokenized with the BERT base tokenizer.
+
+```
+export TASK=msr_ab_sum
+
+export TRAIN_FILE=./data/${TASK}/train.tsv
+export EVAL_FILE=./data/${TASK}/valid.tsv
+export TEST_FILE=./data/${TASK}/test.tsv
+
+export PHRASE_VOCAB_SIZE=500
+export MAX_INPUT_EXAMPLES=1000000
+export OUTPUT_DIR=./outputs/${TASK}
+
+if [ ! -d ${OUTPUT_DIR} ]; then
+    mkdir -p ${OUTPUT_DIR};
+fi
+
+python phrase_vocabulary_optimization.py \
+  --input_file=${TRAIN_FILE} \
+  --vocabulary_size=${PHRASE_VOCAB_SIZE} \
+  --max_input_examples=${MAX_INPUT_EXAMPLES} \
+  --output_file=${OUTPUT_DIR}/label_map.txt
+```
+
+Note that you can also set `MAX_INPUT_EXAMPLES` to a smaller value to get a
+reasonable vocabulary.
+
+### 2. Converting Target Texts to Tags
+
+We've used the 12-layer "BERT-Base, Cased" model for of our experiments.
+Then convert the original TSV datasets into pkl format.
+
+```
+export MAX_SEQ_LENGTH=128
+export LASERTAGGER_CONFIG=./configs/lasertagger_config.json
+
+# Preprocess text to tags
+python lasertagger_preprocessor.py \
+    --train_file=${TRAIN_FILE} \
+    --eval_file=${EVAL_FILE} \
+    --test_file=${TEST_FILE} \
+    --label_map_file=${OUTPUT_DIR}/label_map.txt \
+    --max_seq_length=${MAX_SEQ_LENGTH} \
+    --save_path=${OUTPUT_DIR}
+```
+
+### 3. Model Training
+
+Model hyperparameters are specified in [lasertagger_config.json](configs/lasertagger_config.json). This configuration file extends the original
+`bert_config.json` which comes with the zipped pretrained BERT model.
+
+```
+# Training and evaluation, comment --eval_file_preprocessed to skip evaluation
+python lasertagger_main.py train \
+    --train_file_preprocessed=${OUTPUT_DIR}/lt_train_examples.pkl \
+    --eval_file_preprocessed=${OUTPUT_DIR}/lt_eval_examples.pkl \
+    --test_file_preprocessed=${OUTPUT_DIR}/lt_test_examples.pkl \
+    --label_map_file=${OUTPUT_DIR}/label_map.txt \
+    --max_seq_length=${MAX_SEQ_LENGTH} \
+    --model_config_file=${LASERTAGGER_CONFIG} \
+    --work_dir=${OUTPUT_DIR}/lt
+```
+
+### 4. Prediction and Evaluation
+
+The inference relies on `ROUGE-L` metric which computes the longest common subsequence (LCS) between two pieces of text.
+
+The inference also calculates another metric called `SARI (System output Against References and against the Input sentence)`. It explicitly measures the goodness of words that are added, deleted and kept by the systems.
+
+```
+# Setup ROUGE metrics from https://github.com/google-research/google-research/tree/master/rouge
+pip install rouge-score
```

Comment:
1) if our scripts depend on external packages, then we should add those packages to requirements/requirements_xxx.txt (in this case `xxx` should be `nlp`, I guess...)
2) ROUGE is standard metric and we already depend on e.g. NLTK (in docker? not sure why down there and not in requirements_nlp @okuchaiev can you please take a look at it - because I do not understand it...)

Reply:
Should I add rouge-score to `requirements/requirements_nlp.txt`?

Reply:
@okuchaiev @tkornuta-nvidia I have added `rouge-score` to `requirements/requirements_nlp.txt` in the latest commit.